core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -17840,8 +17840,8 @@ public void deleteGroup(String groupID) throws UserStoreException {
         }
         // #################### </Pre-Listeners> #####################################################
         // Clear cache and mapper tables.
-        groupUniqueIDDomainResolver.removeDomainForGroupId(groupID, UserCoreUtil.extractDomainFromName(groupName),
-                tenantId, false);
+        groupUniqueIDDomainResolver.removeDomainForGroupId(groupID,
+                UserCoreUtil.extractDomainFromName(getGroupNameByGroupId(groupID)), tenantId, false);
```

Comment:
we already have this group name sorted. Why are we invoking this again?

Reply:
Updated with [e005f49](https://github.com/wso2/carbon-kernel/pull/3934/commits/e005f49ced5941312969cf3d377d25f9b05bbcad)